### DISCUSSION TITLE
Prism子界面Region问题

### DISCUSSION THREAD

**Discussion by 17698954505** [BYJRK/DotNet-Discussions#306] [WPF 开发]
在Prism中，在主界面和子界面都存在Region的情况下，如果把函数IsNavigationTarget返回值设置成false（每次导航的时候都是新建界面），第一次导航到子界面没有问题，但是第二次之后都会报子界面的Region已存在，把函数IsNavigationTarget返回值设置成true，则没有问题，原因我大致也明白。
我的问题就是：在子界面设置Region在Prism中是否合理？还有就是Dialog中使用导航是否合理？如果是合理的话，应该用什么思路实现？

**Comment by 17698954505** [comment]
希望各位小伙伴，大佬们能解答一下疑惑

**Comment by BYJRK** [comment]
有没有最小可重现示例代码？

理论上讲这样做是合理的，因为在子界面、对话框中操作主界面的Region（或者说导航）这样的需求是天然存在的。前提是你需要处理好Region服务、窗口及VM的生命周期等。

**Reply by 17698954505** [reply]
感谢解答，确认到了是合理， 我也找到了答案可以通过实现IRegionMemberLifetime接口，设置KeepAlive属性或者使用特性[Region
MemberLifetime(KeepAlive=False)]可以避免Region重复注册的问题.
